robot_smach_states/src/robot_smach_states/world_model.py

```diff
@@ -94,34 +94,87 @@ def execute(self, userdata=None):
         return "done"
 
 
+class UpdateDestEntityPoseWithSrcEntity(smach.State):
+    """ Update the pose of an entity from another entity"""
+
+    def __init__(self, robot, src_entity_designator, dst_entity_designator,
+                 dst_entity_type="waypoint"):
+        """ Constructor
+
+        :param robot: robot object
+        :param src_entity_designator: (EdEntityDesignator) indicating the object from which the pose should be selected
+        :param dst_entity_designator: (EdEntityDesignator) indicating the object of which the pose must be updated
+        :param dst_entity_type: (str) Destination entity type
+        """
+        super(UpdateDestEntityPoseWithSrcEntity, self).__init__(outcomes=["done", "failed"])
+        self._robot = robot
+        ds.check_type(src_entity_designator, Entity)
+        ds.check_type(dst_entity_designator, Entity, str)
+        self._src_entity_designator = src_entity_designator
+        self._dst_entity_designator = dst_entity_designator
+        self._dst_entity_type = dst_entity_type
+
+    def execute(self, userdata=None):
+        """ Looks at the entity and updates its pose using the update kinect service """
+        src_entity = self._src_entity_designator.resolve() if hasattr(
+            self._src_entity_designator,
+            'resolve') else self._src_entity_designator
+
+        dst_entity = self._dst_entity_designator.resolve() if hasattr(
+            self._dst_entity_designator,
+            'resolve') else self._dst_entity_designator
+
+        if (not src_entity) or (not self._robot.ed.get_entity(
+                src_entity.id)) or (not dst_entity):
+            return "failed"
+        else:
+            dst_id = None
+            if isinstance(dst_entity, Entity):
+                dst_id = dst_entity.id
+            else:
+                dst_id = dst_entity
+
+            self._robot.ed.update_entity(id=dst_id,
+                                         frame_stamped=src_entity.pose,
+                                         type=self._dst_entity_type)
+
+            return "done"
+
+
 class SegmentObjects(smach.State):
-    """ Look at an entiy and segment objects within the area desired.
+    """
+    Look at an entity and segment objects within the area desired.
     """
     def __init__(self, robot, segmented_entity_ids_designator, entity_to_inspect_designator,
-                 segmentation_area="on_top_of",
-                 threshold=0.0):
+                 segmentation_area="on_top_of", unknown_threshold=0.0, filter_threshold=0.0):
```

Comment:
Why is this difference necessary @MatthijsBurgh?

Reply:
One is a classification threshold. The other is filtering for which to return.

Reply:
I think the classification threshold should be a constant property of ed. Otherwise we might get some objects in the world model classified with threshold 0.0 and some with threshold 0.3 which means you still dont know with what certainty objects were segmented.

Reply:
Yep, but that up to the user.